### PR TITLE
Web 8064: Consolidate Left & Right areas of Theatre on seat stage

### PR DESCRIPTION
**Pull request by myrepo1010**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-8064
#### What does this PR do?
Update the helper in order to merge left and right seats.
If a merge is needed the code will take the cheapest gold, the cheapest silver and bronze between left and right.
If the prices of the left seat and the right seat is equal than we will pick one of them with a probability of the 50%.
#### Is there anything a developer should be aware of when reviewing this?
You need to switch to WEB-8080 in The Wall in order to see this correctly.
#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [ ] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @taijuten
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [ ] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @owennicol
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [ ] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: helpers/filterSeats.js [hunk 1]
```diff
@@ -58,12 +60,35 @@ module.exports = function(dust) {
       });
     }
 
+    // use Transformer show config to add Gold, Silver or Bronze to packageRate
+    function assignColoursToBands(packageRate) {
+      // assign gold, silver, bronze to packages depending on their current priceBand
+      _.forEach(bandColours, function(sectionValue, sectionKey) {
+        // check for match between transformer and existing data
+        // check for section that might not have full title ( e.g. Grand Circle in Grand Circle (Left) )
+        if (ticketRates[packageRate.ids].section === sectionKey || ticketRates[packageRate.ids].section.indexOf( sectionKey ) > -1 ) {
```
Comment (taijuten) [inline helpers/filterSeats.js]: it looks like you have a lot of `ticketRates[packageRate.ids]`. It might be worth `var`ing this up.

Comment (myrepo1010) [inline helpers/filterSeats.js]: Actually that was Tom's work that I had to merge before working on mine. But yes I think I can refactor that.

File: helpers/filterSeats.js [hunk 2]
```diff
@@ -104,6 +106,152 @@ module.exports = function(dust) {
       });
     }
 
+    function reorderRates(rates, replyIndex) {
+      var bandTypes = {};
+      _.forEach(bandColours, function (sectionObj) {
+        _.forEach(sectionObj, function (quality) {
+          bandTypes[quality] = [];
+        });
+      });
```
Comment (owennicol) [inline helpers/filterSeats.js]: This section could do with a comment to explain where the data is coming from and what we're doing with it here

**Comment by owennicol** [comment]
Looks like this PR has some conflicts

**Pull request merged**